src/services/codefixes/codeFixProvider.ts

```diff
@@ -0,0 +1,53 @@
+/* @internal */
+namespace ts {
+    export interface CodeFix {
+        name: string;
+        errorCodes: string[];
+        getCodeActions(context: CodeFixContext): CodeAction[];
+    }
+
+    export interface CodeFixContext {
+        errorCode: string;
+        sourceFile: SourceFile;
+        span: TextSpan;
+        checker: TypeChecker;
+        newLineCharacter: string;
+    }
+
+    export namespace codeFix {
+        const codeFixes: Map<CodeFix[]> = {};

```

Comment:
does it mean that registration of the fix is global and fix will be visible\available in all loaded projects. I think when we discussed it last time we wanted to make them more fine-grained.